src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -209,9 +217,18 @@ void YARPRobotStatePublisherModule::onRead(JointState &v)
     //        * Add a map string --> indeces
     // Fill the buffer of joints positions
     const iDynTree::Model& model = m_kinDynComp.model();
+    iDynTree::JointIndex jntIndex;
     for (size_t i=0; i < v.name.size(); i++)
     {
-        iDynTree::JointIndex jntIndex = model.getJointIndex(v.name[i]);
+        for (size_t j=0; j < model.getNrOfDOFs(); j++)
+        {
+            if (v.name[i] == model.getJointName(j))
+            {
+                jntIndex = model.getJointIndex(v.name[i]);
+                break;
+            }
+        }
```

Comment:
If am quite confused by this logic. What is the rationale for that? If you want to make sure that a given joint in the iDynTree models has a number of dofs that are equal to 1, you can just call, 
`model.getJoint(jntIndex)->getNrOfDOFs()` and check if it is equal to 1.

Reply:
Here, I guess we are checking the consistency between the JointState message joints and the Model joints.

Aside: I'd suggest to be more verbose with the variable names to make it more readable.

Reply:
> I guess we are checking the consistency between the JointState message joints and the Model joints

If a joint name in the jointState messag is not found in the model, `getJointIndex` return `JOINT_INVALID_INDEX ` which was already taken into account in the following check (https://github.com/robotology/idyntree/pull/538/files/8cd4a1b3cd8a1ada7859cb859eb31dc04310f528#diff-c2170ce79e08829b7556ba58dcbca3a3L215).

Honestly, I am a bit lost too in understanding the purpose here. For how I see it, we don't need this check.

Reply:
> Honestly, I am a bit lost too in understanding the purpose here. For how I see it, we don't need this check.

Then I think we can remove it.

Reply:
Done https://github.com/robotology/idyntree/pull/538/commits/c15230ac792c72523668abc4506510d862748224